client/src/extension.ts

```diff
@@ -89,19 +89,70 @@ export function activate(context: ExtensionContext) {
         ExecYseopCliCommand(yseopCliPath, 'info');
     });
 
-    // Create the language client and start the client.
-    const disposable = new LanguageClient(
+    // Create the language client.
+    const languageClient = new LanguageClient(
         'yseopml',
         'Yseop Markup Language language server',
         serverOptions,
         clientOptions,
         true,
-    ).start();
+    );
+    // Start the language client.
+    const disposable = languageClient.start();
 
     // Push the disposable to the context's subscriptions so that the
     // client can be deactivated on extension deactivation.
     // Also register the custom commands.
     context.subscriptions.push(disposable, batchCmd, compileCmd, testCmd, cleanCmd, cleanallCmd, packageCmd, infoCmd);
+
+    let parseAllProjectFilesAtStartup: boolean = workspace.getConfiguration('yseopml').get('parseAllProjectFilesAtStartup');
+
+    workspace.onDidChangeConfiguration((event) => {
+        if (event.affectsConfiguration('yseopml.parseAllProjectFilesAtStartup')) {
+            parseAllProjectFilesAtStartup = workspace.getConfiguration('yseopml').get('parseAllProjectFilesAtStartup');
+        }
+    });
+
+    if (!parseAllProjectFilesAtStartup) {
+        // nothing more to do
+        return;
+    }
+    /*
+        * List of all the yseopml file extensions known by this extension as set in `client/package.json`.
+        */
+    const yseopmlExtensions = [
+        'kao',
+        'yclass',
+        'ytextfunction',
```

Comment:
I'm not entirely sure you should parse XML files like `.ytextfunction`. Better focus on files that are in YML, don't you think?
Otherwise, why not adding rule files?

Reply:
Maybe using two options:
- only YML files
- all Yseop studio files and YML files

Reply:
Good point. Maybe the extension cannot even obtain useful information from XML-base files.

Reply:
In fact, parsing Yseop studio files works more or less, but is indeed not quite so useful.
I think I will just remove the parsing of these files. Maybe it will encourage people to use the new `convert` feature?